source/base/parameter_handler.cc

```diff
@@ -1862,6 +1862,7 @@ namespace
 
 bool ParameterHandler::read_input_from_xml (std::istream &in)
 {
+  AssertThrow(in, ExcIO());
```

Comment:
Yes, I'm in favor of deprecating these functions (well, actually just removing the return value).

I have no idea how many people use the return value, but it's certainly confusing because we don't use return values anywhere else in the library. I think there is virtue in uniform style.

I opened #2984 for this.